echopype/echodata/combine.py

```diff
@@ -0,0 +1,131 @@
+import warnings
+from datetime import datetime
+from typing import List
+
+import xarray as xr
+from _echopype_version import version as ECHOPYPE_VERSION
+
+from ..core import SONAR_MODELS
+from ..qc import coerce_increasing_time, exist_reversed_time
+from .echodata import EchoData
+
+
+def union_attrs(datasets: List[xr.Dataset]) -> List[xr.Dataset]:
+    """
+    Merges attrs from a list of datasets.
+    Prioritizes keys from later datsets.
+    """
+
+    total_attrs = dict()
+    for ds in datasets:
+        total_attrs.update(ds.attrs)
+    for ds in datasets:
+        ds.attrs = total_attrs
+    return datasets
+
+
+def assemble_combined_provenance(input_paths):
+    return xr.Dataset(
+        data_vars={
+            "src_filenames": ("file", input_paths),
+        },
+        attrs={
+            "conversion_software_name": "echopype",
+            "conversion_software_version": ECHOPYPE_VERSION,
+            "conversion_time": datetime.utcnow().isoformat(timespec="seconds")
+            + "Z",  # use UTC time
+        },
+    )
+
+
+def combine_echodata(echodatas: List[EchoData]) -> EchoData:
+    """
+    Combines multiple echodata objects into a single EchoData by combining
+        their groups individually.
+
+    If echodatas contain EchoData objects with different or None sonar_models,
+        a ValueError is raised.
+    """
```

Comment:
Since this method will be public and used by users, could you expand the docstring to follow the numpy doc. Here's a documentation about it: https://numpydoc.readthedocs.io/en/latest/format.html